azkaban-common/src/main/java/azkaban/container/models/AzKubernetesV1PodTemplate.java

```diff
@@ -71,7 +73,17 @@ public static synchronized AzKubernetesV1PodTemplate getInstance(String template
    * @return the {@link V1PodSpec} POD spec generated from the template.
    * @throws IOException If unable to read the template file.
    */
-  public synchronized V1PodSpec getPodSpecFromTemplate() throws IOException {
+  public V1PodSpec getPodSpecFromTemplate() throws IOException {
```

Comment:
This method doesn't need to be synchronized anymore?

Reply:
I believe this code isn't called from different threads and we're just reading the data so synchronized isn't needed.